### PR TITLE
puppeteer: Added 'domcontentloaded' page event which works on page.on() callbacks.

### PR DESCRIPTION
**Pull request by nasikshafeek**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#event-domcontentloaded
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/puppeteer/puppeteer-tests.ts [hunk 1]
```diff
@@ -502,3 +502,20 @@ puppeteer.launch().then(async browser => {
...
+(async () => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+  page.on('domcontentloaded', async () => {
+    page.evaluate(() => {
+      console.log('dom changed');
+    });
+  });
+
+  try {
```
Comment (SimonSchick) [inline types/puppeteer/puppeteer-tests.ts]: I think this code chunk is redundant for this test.

File: types/puppeteer/index.d.ts [hunk 2]
```diff
@@ -1272,6 +1272,11 @@ export interface PageEventObj {
    * Puppeteer can respond to the dialog via Dialog's accept or dismiss methods.
    */
   dialog: Dialog;
+  /**
+   * Emitted when the initial HTML document has been completely loaded and parsed, 
+   * without waiting for stylesheets, images, and subframes to finish loading.
+   */
+  domcontentloaded: undefined;
```
Comment (SimonSchick) [inline types/puppeteer/index.d.ts]: I think `never` or `void` might be better here actually.

Comment (kisanme) [reply types/puppeteer/index.d.ts]: Agreed! `void` will be better, will replace this.

File: types/puppeteer/puppeteer-tests.ts (hunk 1 above)
Comment (kisanme) [reply types/puppeteer/puppeteer-tests.ts]: @SimonSchick did you mean to remove `try-catch` and keep the `page.goto('.....` line?

Comment (SimonSchick) [reply types/puppeteer/puppeteer-tests.ts]: Yes, these tests are meant to test type safety, not functionality.

**Comment by kisanme** [comment]
Thanks @SimonSchick for the information. Fixed the PR according to your comments.

**Comment by SimonSchick** [comment]
Uhh, did you mean to close this?

**Comment by kisanme** [comment]
err, nope. Will reopen this.

**Comment by kisanme** [comment]
Used `never` instead of `void` as it wasn't allowed as per the standards (the [build failed](https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/479214627?utm_source=github_status&utm_medium=notification) actually)

@SimonSchick can you please review this?

**Review by rbuckton — APPROVED** [review]

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**